### ISSUE TITLE
Port forwarding

### ISSUE THREAD

**Issue by brandonwillard** [chrisbarrett/kubernetes-el#122]
It would be nice if this package had a port-forwarding option/feature.

**Comment by noorul** [comment]
This package is a wrapper around kubectl
. Is this feature request part of kibectl command.

**Comment by brandonwillard** [comment]
Yes, see [`kubectl port-forward`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#port-forward).

**Comment by noorul** [comment]
I am not. @jinnovation ?

**Comment by jinnovation** [comment]
Not at all. @kavirajk: all yours if you've like to take a stab at it.

**Comment by jinnovation** [comment]
Carrying this work out will require some groundwork first—namely, in the `kubernetes-kubectl` function.

`kubernetes-kubectl` was designed for one-off, short-lived invocations, like the ones used to query resources, e.g. `kubectl pods` et al. Port-forwarding, meanwhile, is long-lived, and the port-forward should be kept active until the user explicitly cancels it. Furthermore, users could request *multiple* port-forwards to run at once.

In its current form, `kubernetes-kubectl` is insufficiently flexible to use for port-forwarding, for at minimum the following reasons:
- Its hard-coding of the corresponding process's output and stderr buffers directly precludes multiple invocations from running at once;
- Likewise its hard-coding of the process name itself;
- It assumes that general options, e.g. `--namespace`, will only ever come from `(kubernetes-state)`, whereas ideally we'd like to permit users to port-forward from any namespace and any context, regardless of what namespace/context pair they're currently browsing in the overview.

All of these will have to change in order for us to effectively implement port-forwarding. Luckily, these are all pretty discrete and explicit improvements.

**Comment by jinnovation** [comment]
#256 introduces the [`kubernetes--ported-process-record`](https://github.com/kubernetes-el/kubernetes-el/blob/master/kubernetes-process.el#L25-L38) class that might be useful for port-forward process management as well.
